xarray/core/variable.py

```diff
@@ -269,9 +269,9 @@ def as_compatible_data(
 
     Finally, wrap it up with an adapter if necessary.
     """
-    if fastpath and getattr(data, "ndim", 0) > 0:
-        # can't use fastpath (yet) for scalars
-        return cast("T_DuckArray", _maybe_wrap_data(data))
+    ndim = getattr(data, "ndim", None)
```

Comment:
Small point relative to avoiding `_maybe_wrap_data`: 

Wouldn't we only want to get `ndim` if `fastpath is True`? So we could only change the comparison to `0` to a comparison to `None` on L272?

Reply:
Thank you for the thorough and quick review. you are correct.

Reply:
Looks like both `pd.Index` and `ExtensionArray` define `ndim` so should be OK. It'd be nice to add this as a comment.